### DISCUSSION TITLE
Nicknames (bioimageio managed ids) for non-model resource descriptions

### DISCUSSION THREAD

**Discussion by FynnBe** [bioimage-io/collection-bioimage-io#660] [Ideas]
With https://github.com/bioimage-io/collection-bioimage-io/issues/659 giving an additional incentive to come up with bioimageio internal ids for non-model resources I gave it another thought and collected this overview:

For context: We use animal nicknames (positive adjective + animal for which a unicode character exists) to identify models. Here are some ideas how to extend that to our other resource types.

## delicious-food, (for datasets?!? because animals eat food!)
e.g tasty-burger

cheese 🧀
egg 🥚 
pancakes 🥞
waffle 🧇
butter 🧈
bread 🍞
croissant 🥐
baguette 🥖
pretzel 🥨
bagel 🥯 

burger 🍔
french-fries 🍟
hot-dog 🌭
pizza 🍕
sandwich 🥪 
taco 🌮 
burrito 🌯 
kebab 🥙
falafel 🧆

pasta 🍝
curry 🍛 
paella 🥘
fried-shrimp 🍤  (so close to be an animal) 
salad 🥗

bacon 🥓
knuckle 🍖
drumstick 🍗
steak 🥩

takeout 🥡
dumpling 🥟
rice 🍚
fortune-cookie 🥠
moon-cake 🥮
sushi 🍣
bento(box) 🍱 

cake 🍰
cupcake 🧁
pie 🥧

sweets
doughnut 🍩
cookie 🍪
popcorn 🍿
custard 🍮
honey 🍯

ice-cream 🍨
chocolate 🍫
candy 🍬
lollipop 🍭

fruits
🍇🍑🍒🍓🥝🍉🍈🍊🍋🍌🍍🥭🥥🍎🍏🍐
vegetables
🥔🍠🥜🥬🥦🍄🍅🥕🌶🧄🧅🍆🥒🥑

35+13+15+14 = 77

skipped, not rendered for me
🫘🫘🫑🫒🫐🫓🫑🫒

skipped, unclear name
🍳🍘🍙🍥🍡🍢🍜🥫🍲🎂🍦🍧

### fast-vehicles (for notebooks??)
e.g. fast-bike, impressive-scooter
(<58) 🚲🛴🛵🛺🏍🚕🚖🚗🚘🚔🚓🚙🚑🚒⛟🛻🚚🚛🚏🚌🚍🚎🚐🚉🚂🚆🚄🚅🚝🚃🚇🚈🚊🚋🚞🛤🚟🚠🚡🏎🚜⛴🚢⛵🚣🚤🛥🛫🛬🛩✈🛧🛪🛨🛦🚁🚀🛸

## number-currency
e.g. 100-bitcoin ₿
bitcoin ₿
dollar $💵
peso $💲
cent ¢
euro €💶
pound £💷
yen ¥💴
rupee ₨৳₹
cedi ₵
colon ₡
baht ฿
franc ₣
... ₲₭₥₦₱₽₴₮₩₢₫₯₪₧₰៛
money 💰
credit 💳

14 + (0--16)

skipped, outdated: ₳

skipped, right to left/weird spacing: ؋﷼

# Other unicode symbol groups
### drinks
(17), maybe inappropriate?, maybe add some into 'food'  if we need more there

### instruments
(6)🎹🎻🎷🎺🎸🪕🪗🥁 

### playing cards
(57) 🂱🂲🂳🂴🂵🂶🂷🂸🂹🂺
🂻🂼🂽🂾
🂡🂢🂣🂤🂥🂦🂧🂨🂩🂪
🂫🂬🂭🂮
🃁🃂🃃🃄🃅🃆🃇🃈🃉🃊
🃋🃌🃍🃎
🃑🃒🃓🃔🃕🃖🃗🃘🃙🃚
🃛🃜🃝🃞
🃏

### chess pieces (maybe not)
e.g. bold-black-king
(12) ♚ ♛ ♜ ♝ ♞ ♟ ♔ ♕ ♖ ♗ ♘ ♙

### buildings
harbor⚓
hospital 🏥
house 🏠
bridge 🌉
construction(-site) 🏗
...

### games/sport
soccer ⚽
chess 🙿
checkers 🙾
puzzle 🧩
billiards 🎱
darts 🎯
handball 🤾
...⚾🥎🏀🏐🏈🏉🏏🏑🏒🥅🥍
🏓🎾🏸🥊🥋🤺🤼🏃🧗🏇🏋🏹🤸🤹🛹🛼🥏🎳
🏊🏄🤽🛶
🎿⛸⛷🏂🛷🥌
🎣🪂🏌
mahjong (one of 🀇🀈🀉🀊🀋🀌🀍🀎🀏🀐🀑🀒🀓🀔🀕🀖🀗🀘🀙🀚🀛🀜🀝🀞🀟🀠🀡🀀🀁🀂🀃🀢🀣🀤🀥🀦🀧🀨🀩🀄🀅🀆🀪)

# Our other types
`application` e.g. ilastik/ilastik should probably keep their id and have their own (non-unicode) symbol... these don't scale anyway.
We could consider the instruments for more generic applications.

We should also already think of nickname schemes for `function` (or processing or generic processing or ... name tbd) and `workflow` (or pipeline or.... name tbd).
Maybe functions as number-currency and workflows buidings/infrastructure?
Or functions as playing cards/chess pieces, etc and workflows as games/sport

**Comment by oeway** [comment]
Hey, thanks for the great initiative! 

I like the idea of food! But maybe we can use animal food(plants) rather than human food (cheese) for data, since we need to feed animals with their corresponding food. Say we can only feed cows some types of grass. Although it maybe more complicated to find out what does a certain animal eat. I would perhaps not use a fixed list, give the right for the user to select, or use an LLM (e.g. we can run a small llama2 model in the bioengine to provide such service, or just the bioimageio chatbot) in the CI to generate food from animal for the selected animal.

For applications, we have their own name, most likely, so we won't need to generate. Or some island or something where we breed the animals or gather the animal.

**Reply by FynnBe** [reply]
As the animal names are randomly generated I don't think there is any hope to match them with suitable food. 
Available plant-like characters are also somewhat limited. 
I found 16 that would maybe fit the plant category: 🌵🌴🍀🍁🍃🌱🌽🌾🌰🌿🌷🌹🌺🌻🌼🥀
These are not exactly intuitive animal food items, so I would prefer to lift the "food" category one abstraction level, otherwise I'm afraid we are pushing the metaphor a bit too far.

How about adding these to "food"?
corn 🌽
clover 🍀
chestnut 🌰
herb 🌿

**Reply by cfusterbarcelo** [reply]
I really like this initiative of having this food for animals. Could we add some adjectives too? But ones dedicated for food.

Some could be:
Delicious
Tasty
Savory
Yummy
Sticky
Bitter
Crispy
Tender
Juicy
Spicy
...

**Reply by FynnBe** [reply]
yes, combining it with a postive adjective that is somewhat suitable for food is the idea. Otherwise we won't have enough options to use this as an identifier 👍

**Reply by k-dominik** [reply]
IMO we do not really need to restrict ourselves to foods commonly given to animals. It's not that we took care choosing only adjectives that fit animals before, in the case of the models. For me the ones where there was some dissonance were easier to remember anyway (`affable shark`...).

**Reply by esgomezm** [reply]
noice! BUT not all animals are veggie + we have models to process plant biology data, so I'm not sure whether this would be a bit awkward. Besides the plants, why not include fruit and vegetables in general?

**Reply by FynnBe** [reply]
> For me the ones where there was some dissonance were easier to remember anyway (affable shark...).

same, and funnier! We shouldn't loose our humor... I like the idea of the affable shark enjoying a bitter ice cream 🍨🦈

> Besides the plants, why not include fruit and vegetables in general?

I would include them as "food", I only put them there as groups, because I felt it a bit redundant to spell out all the names... they should almost all be usable (maybe only one apple (and not a red and a green one).
fruits
🍇🍑🍒🍓🥝🍉🍈🍊🍋🍌🍍🥭🥥🍎🍏🍐
vegetables
🥔🍠🥜🥬🥦🍄🍅🥕🌶🧄🧅🍆🥒🥑

**Comment by FynnBe** [comment]
another category could be "clothes" with 42 usable symbols:
👙 bikini
🩱 swimsuit
👗 dress
👚 blouse
🎀 ribbon
👘 kimono
🥻 sari
👒 hat
👠 high-heels
🩰 ballet-shoes
👡 sandal
👢 boot
🥿 flat-shoe
👛 purse
👜 handbag
🛍 shopping-bag
👝 pouch
👔 necktie
👕 t-shirt
🎽 running-shirt
👖 jeans
🩲 briefs
🩳 shorts
👞 shoe
👟 sneaker
🥾 hiking-boot
🧣 scarf
🧤 gloves
🧥 coat
🧦 socks
🧢 cap 
👑 crown
🎩 top-hat
🎓 graduation-hat
⛑ helmet
🥼 lab-coat
🦺 safety-vest 
🎒 backpack (satchel)
👓 glasses
🕶 sunglasses
🥽 googles
🤿 diving-mask

🩴🪖 (doesn't render for me)

there are more with people in them, but I don't think that would fit nicely in our zoo...
🤵 tuxedo
👰 veil
👮 uniform
👳 turban
🧕 headscarf
🤠 cowboy-hat
...

**Comment by FynnBe** [comment]
the vehicle category could be named like this with 44 usable symbols:
🚲 bicycle
🛴 kick-scooter
🛵 motor-scooter
🛺 tuk-tuk
🏍 motorcycle
🚕 taxi
🚗 car
🚓 police-car
🚙 suv
🚑 ambulance
🚒 fire-engine
⛟ truck
🚚 delivery-truck
🚛 lorry
🚌 bus
🚎 trolleybus
🚐 minibus
🚂 locomotive
🚆 train
🚄 high-speed-train
🚅 bullet-train
🚝 monorail-train
🚃 railway-wagon
🚈 light-rail-vehicle
🚇 metro-train
🏎 racing-car
🚋 tram
🚠 gondola
🚟 suspension-train
🚞 mountain-train
🚜 tractor
⛴ ferry
🚢 ship
⛵ sailboat
🚣 rowing-boat
🚤 speedboat
🛥 motor-boat
🛩 small-airplane
✈ jumbo-jet
🛪 airplane
🛦 jet-fighter
🚁 helicopter
🚀 rocket
🛸 flying-saucer


🛻 pickup
🛫 airplane departure
🛬 airplane arrival
🚡 aerial-tramway

**Comment by FynnBe** [comment]
general question: Should we reuse symbols, i.e. use 🚒 fire-engine and 🚒 fire-truck ?

**Reply by FynnBe** [reply]
for now I's suggest to refrain from reusing symbols... if we'll ever run out of options we could resort to this...
